specification/types.md

```diff
@@ -123,8 +127,18 @@ A structure containing the following fields:
 
 A structure which supports definition of arbitrary properties, with keys of type `string`, and values of type `boolean`, `string`, or `number`.
 
+`key` is a required property with a value of type `string`.
+`type` is an optional property with a value of type `string`, indicating which client resolution method should be used to resolve the flag key.
```

Comment:
I don't think this is spelled out, but we might want to.   `key` and `type` are reserved properties on the Flag Metadata object.